cloudinit/config/cc_apk_configure.py

```diff
@@ -0,0 +1,264 @@
+                        settings specified in cloud config. To disable this
+                        behavior and preserve the repositories file from the
+                        pristine image, set ``preserve_repositories``
+                        to ``true``.
+
+                        The ``preserve_repositories`` option overrides
+                        all other config keys that would alter
+                        ``/etc/apk/repositories``.
+                    """)
+                },
+                'alpine_repo': {
+                    'type': ['object', 'null'],
+                    'properties': {
+                        'base_url': {
+                            'type': 'string',
+                            'default': DEFAULT_MIRROR,
+                            'description': dedent("""\
+                                The base URL of an Alpine repository, or
+                                mirror, to download official packages from.
+                                If not specified then it defaults to ``{}``
+                            """.format(DEFAULT_MIRROR))
+                        },
+                        'community_enabled': {
+                            'type': 'boolean',
+                            'default': False,
+                            'description': dedent("""\
+                                Whether to add the Community repo to the
+                                repositories file. By default the Community
+                                repo is not included.
+                            """)
+                        },
+                        'testing_enabled': {
+                            'type': 'boolean',
+                            'default': False,
+                            'description': dedent("""\
+                                Whether to add the Testing repo to the
+                                repositories file. By default the Testing
+                                repo is not included. It is only recommended
+                                to use the Testing repo on a machine running
+                                the ``Edge`` version of Alpine as packages
+                                installed from Testing may have dependancies
+                                that conflict with those in non-Edge Main or
+                                Community repos."
+                            """)
+                        },
+                        'version': {
+                            'type': 'string',
+                            'description': dedent("""\
+                                The Alpine version to use (e.g. ``v3.12`` or
+                                ``edge``)
+                            """)
+                        },
+                    },
+                    'required': ['version'],
+                    'minProperties': 1,
+                    'additionalProperties': False,
+                },
+                'local_repo_base_url': {
```

Comment:
Have you considered using multiple repositories at this point? I am able to imagine that one organization can have multiple repositories managed by different teams (repository provided by IT security for quick security patches, project team repository with specific product releases etc.).

Reply:
I hadn't thought about that.

However additional repositories could be added once the initial /etc/apk/repositories file is set up by cc_apk_configure, whether via cloud-init cc_runcmd or cc_write_files or some other means. It's hard to attempt to cater for all possible scenarios.